// ngx-stripe/esm2015/lib/services/stripe-instance.interface.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function StripeServiceInterface() { }
if (false) {
    /**
     * @return {?}
     */
    StripeServiceInterface.prototype.getInstance = function () { };
    /**
     * @param {?=} options
     * @return {?}
     */
    StripeServiceInterface.prototype.elements = function (options) { };
    /**
     * @param {?} a
     * @param {?} b
     * @return {?}
     */
    StripeServiceInterface.prototype.createToken = function (a, b) { };
    /**
     * @param {?} a
     * @param {?=} b
     * @return {?}
     */
    StripeServiceInterface.prototype.createSource = function (a, b) { };
    /**
     * @param {?} source
     * @return {?}
     */
    StripeServiceInterface.prototype.retrieveSource = function (source) { };
    /**
     * @param {?} options
     * @return {?}
     */
    StripeServiceInterface.prototype.paymentRequest = function (options) { };
    /**
     * @param {?} a
     * @return {?}
     */
    StripeServiceInterface.prototype.handleCardAction = function (a) { };
    /**
     * @param {?} a
     * @param {?=} b
     * @param {?=} c
     * @return {?}
     */
    StripeServiceInterface.prototype.handleCardPayment = function (a, b, c) { };
    /**
     * @param {?} a
     * @param {?=} b
     * @return {?}
     */
    StripeServiceInterface.prototype.confirmPaymentIntent = function (a, b) { };
    /**
     * @param {?} a
     * @param {?} b
     * @param {?=} c
     * @return {?}
     */
    StripeServiceInterface.prototype.createPaymentMethod = function (a, b, c) { };
    /**
     * @param {?} a
     * @param {?=} b
     * @param {?=} c
     * @return {?}
     */
    StripeServiceInterface.prototype.handleCardSetup = function (a, b, c) { };
}
