// ngx-stripe/esm5/lib/interfaces/payment-intent.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function HandleCardPaymentOptions() { }
if (false) {
    /** @type {?|undefined} */
    HandleCardPaymentOptions.prototype.payment_method;
    /** @type {?|undefined} */
    HandleCardPaymentOptions.prototype.payment_method_data;
    /** @type {?|undefined} */
    HandleCardPaymentOptions.prototype.shipping;
    /** @type {?|undefined} */
    HandleCardPaymentOptions.prototype.receipt_email;
    /** @type {?|undefined} */
    HandleCardPaymentOptions.prototype.save_payment_method;
}
/**
 * @param {?} handleCardPaymentOptions
 * @return {?}
 */
export function isHandleCardPaymentOptions(handleCardPaymentOptions) {
    return 'type' in handleCardPaymentOptions;
}
/**
 * @record
 */
export function ConfirmPaymentIntentOptions() { }
if (false) {
    /** @type {?|undefined} */
    ConfirmPaymentIntentOptions.prototype.return_url;
    /** @type {?|undefined} */
    ConfirmPaymentIntentOptions.prototype.use_stripe_sdk;
    /** @type {?|undefined} */
    ConfirmPaymentIntentOptions.prototype.source;
}
/**
 * @record
 */
export function PaymentIntent() { }
if (false) {
    /** @type {?} */
    PaymentIntent.prototype.id;
    /** @type {?} */
    PaymentIntent.prototype.object;
    /** @type {?} */
    PaymentIntent.prototype.amount;
    /** @type {?} */
    PaymentIntent.prototype.amount_capturable;
    /** @type {?} */
    PaymentIntent.prototype.amount_received;
    /** @type {?} */
    PaymentIntent.prototype.application;
    /** @type {?} */
    PaymentIntent.prototype.application_fee_amount;
    /** @type {?} */
    PaymentIntent.prototype.canceled_at;
    /** @type {?} */
    PaymentIntent.prototype.cancellation_reason;
    /** @type {?} */
    PaymentIntent.prototype.capture_method;
    /** @type {?} */
    PaymentIntent.prototype.charges;
    /** @type {?} */
    PaymentIntent.prototype.client_secret;
    /** @type {?} */
    PaymentIntent.prototype.confirmation_method;
    /** @type {?} */
    PaymentIntent.prototype.created;
    /** @type {?} */
    PaymentIntent.prototype.currency;
    /** @type {?} */
    PaymentIntent.prototype.customer;
    /** @type {?} */
    PaymentIntent.prototype.description;
    /** @type {?} */
    PaymentIntent.prototype.last_payment_error;
    /** @type {?} */
    PaymentIntent.prototype.livemode;
    /** @type {?} */
    PaymentIntent.prototype.metadata;
    /** @type {?} */
    PaymentIntent.prototype.next_source_action;
    /** @type {?} */
    PaymentIntent.prototype.on_behalf_of;
    /** @type {?} */
    PaymentIntent.prototype.payment_method;
    /** @type {?} */
    PaymentIntent.prototype.payment_method_types;
    /** @type {?} */
    PaymentIntent.prototype.receipt_email;
    /** @type {?} */
    PaymentIntent.prototype.review;
    /** @type {?} */
    PaymentIntent.prototype.shipping;
    /** @type {?} */
    PaymentIntent.prototype.source;
    /** @type {?} */
    PaymentIntent.prototype.statement_descriptor;
    /** @type {?} */
    PaymentIntent.prototype.status;
    /** @type {?} */
    PaymentIntent.prototype.transfer_data;
    /** @type {?} */
    PaymentIntent.prototype.transfer_group;
}
/**
 * @record
 */
export function PaymentIntentResult() { }
if (false) {
    /** @type {?|undefined} */
    PaymentIntentResult.prototype.paymentIntent;
    /** @type {?|undefined} */
    PaymentIntentResult.prototype.error;
}
/**
 * @record
 */
export function PaymentMethodData() { }
if (false) {
    /** @type {?|undefined} */
    PaymentMethodData.prototype.type;
    /** @type {?|undefined} */
    PaymentMethodData.prototype.billing_details;
    /** @type {?|undefined} */
    PaymentMethodData.prototype.card;
    /** @type {?} */
    PaymentMethodData.prototype.metadata;
}
/**
 * @param {?} paymentMethodData
 * @return {?}
 */
export function isPaymentMethodData(paymentMethodData) {
    return paymentMethodData && 'type' in paymentMethodData;
}
/**
 * @record
 */
export function PaymentMethod() { }
if (false) {
    /** @type {?} */
    PaymentMethod.prototype.id;
    /** @type {?} */
    PaymentMethod.prototype.object;
    /** @type {?|undefined} */
    PaymentMethod.prototype.billing_details;
    /** @type {?} */
    PaymentMethod.prototype.card;
    /** @type {?} */
    PaymentMethod.prototype.created;
    /** @type {?} */
    PaymentMethod.prototype.customer;
    /** @type {?} */
    PaymentMethod.prototype.livemode;
    /** @type {?} */
    PaymentMethod.prototype.metadata;
    /** @type {?} */
    PaymentMethod.prototype.type;
}
/**
 * @record
 */
export function PaymentMethodResult() { }
if (false) {
    /** @type {?|undefined} */
    PaymentMethodResult.prototype.paymentMethod;
    /** @type {?|undefined} */
    PaymentMethodResult.prototype.error;
}
