// ngx-stripe/esm2015/lib/interfaces/stripe.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InjectionToken } from '@angular/core';
/** @type {?} */
export const STRIPE_PUBLISHABLE_KEY = new InjectionToken('Stripe Publishable Key');
/** @type {?} */
export const STRIPE_OPTIONS = new InjectionToken('Stripe Options');
/**
 * @record
 */
export function StripeJS() { }
if (false) {
    /**
     * @param {?=} options
     * @return {?}
     */
    StripeJS.prototype.elements = function (options) { };
    /**
     * @param {?} el
     * @param {?=} cardData
     * @return {?}
     */
    StripeJS.prototype.createToken = function (el, cardData) { };
    /**
     * @param {?} account
     * @param {?} accountData
     * @return {?}
     */
    StripeJS.prototype.createToken = function (account, accountData) { };
    /**
     * @param {?} account
     * @param {?} bankAccountData
     * @return {?}
     */
    StripeJS.prototype.createToken = function (account, bankAccountData) { };
    /**
     * @param {?} pii
     * @param {?} piiData
     * @return {?}
     */
    StripeJS.prototype.createToken = function (pii, piiData) { };
    /**
     * @param {?} el
     * @param {?=} sourceData
     * @return {?}
     */
    StripeJS.prototype.createSource = function (el, sourceData) { };
    /**
     * @param {?} sourceData
     * @return {?}
     */
    StripeJS.prototype.createSource = function (sourceData) { };
    /**
     * @param {?} source
     * @return {?}
     */
    StripeJS.prototype.retrieveSource = function (source) { };
    /**
     * @param {?} options
     * @return {?}
     */
    StripeJS.prototype.paymentRequest = function (options) { };
    /**
     * @param {?} clientSecret
     * @param {?} cardElement
     * @param {?=} data
     * @return {?}
     */
    StripeJS.prototype.handleCardPayment = function (clientSecret, cardElement, data) { };
    /**
     * @param {?} clientSecret
     * @param {?=} data
     * @return {?}
     */
    StripeJS.prototype.handleCardPayment = function (clientSecret, data) { };
    /**
     * @param {?} clientSecret
     * @return {?}
     */
    StripeJS.prototype.handleCardAction = function (clientSecret) { };
    /**
     * @param {?} clientSecret
     * @param {?=} data
     * @return {?}
     */
    StripeJS.prototype.confirmPaymentIntent = function (clientSecret, data) { };
    /**
     * @param {?} type
     * @param {?} cardElement
     * @param {?=} data
     * @return {?}
     */
    StripeJS.prototype.createPaymentMethod = function (type, cardElement, data) { };
    /**
     * @param {?} clientSecret
     * @param {?} element
     * @param {?=} data
     * @return {?}
     */
    StripeJS.prototype.handleCardSetup = function (clientSecret, element, data) { };
    /**
     * @param {?} clientSecret
     * @param {?=} data
     * @return {?}
     */
    StripeJS.prototype.handleCardSetup = function (clientSecret, data) { };
}
/**
 * @record
 */
export function Options() { }
if (false) {
    /** @type {?|undefined} */
    Options.prototype.stripeAccount;
}
